### PR TITLE
Added ACS Chat preview feature: introducing chat data retention as a thread property.

### PR DESCRIPTION
**Pull request by angiurgiu**
This preview API introduces introducing chat data retention as a thread property.

### CODE REVIEW

File: sdk/communication/Azure.Communication.Chat/src/ChatClient.cs [hunk 1]
```diff
@@ -62,7 +62,17 @@ public virtual async Task<Response<CreateChatThreadResult>> CreateChatThreadAsyn
...
+                var createChatThreadOptions = new CreateChatThreadOptions(topic)
+                {
+                    IdempotencyToken = idempotencyToken,
+                };
+
+                foreach (var value in participants.ToList())
+                {
+                    createChatThreadOptions.Participants.Add(value);
+                }
+
```
Comment (LuChen-Microsoft) [inline sdk/communication/Azure.Communication.Chat/src/ChatClient.cs:74]: Why do we need changes here? Seems the `ToChatParticipantInternal()` is removed from the function.

Comment (LuChen-Microsoft) [reply sdk/communication/Azure.Communication.Chat/src/ChatClient.cs:74]: Oh, I see the refactor in the ChatRestClient

File: sdk/communication/Azure.Communication.Chat/src/autorest.md [hunk 2]
```diff
@@ -11,9 +11,10 @@ If any of the new objects needs to be overwritten, add the required changes to t
 
 ``` yaml
 title: Chat
-tag: package-2023-11-15-preview
+tag: package-2024-03-15-preview
 model-namespace: false
 require:
-    -  https://raw.githubusercontent.com/Azure/azure-rest-api-specs/00c56827fe8064415b2b378b5941a833b22f5236/specification/communication/data-plane/Chat/readme.md
+    -  https://raw.githubusercontent.com/Azure/azure-rest-api-specs/fe375c4b0f59b587a263e4b5754925ccc660bd47/specification/communication/data-plane/Chat/readme.md
```
Comment (LuChen-Microsoft) [inline sdk/communication/Azure.Communication.Chat/src/autorest.md:17]: The API spec is already merged to the main branch. There's no change of the content,
https://[raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/communication/data-plane/Chat/preview/2024-03-15-preview/communicationserviceschat.json](https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/communication/data-plane/Chat/preview/2024-03-15-preview/communicationserviceschat.json)

File: sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs [hunk 3]
```diff
@@ -204,6 +210,7 @@ public partial class ChatThreadProperties
         public System.DateTimeOffset? DeletedOn { get { throw null; } }
         public string Id { get { throw null; } }
         public System.Collections.Generic.IReadOnlyDictionary<string, string> Metadata { get { throw null; } }
+        public Azure.Communication.Chat.ChatRetentionPolicy RetentionPolicy { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (angiurgiu) [reply sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs:213]: No, I will remove it

File: sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs [hunk 4]
```diff
@@ -249,6 +284,7 @@ public partial class UpdateChatThreadPropertiesOptions
     {
         public UpdateChatThreadPropertiesOptions() { }
         public System.Collections.Generic.IDictionary<string, string> Metadata { get { throw null; } }
+        public Azure.Communication.Chat.ChatRetentionPolicy RetentionPolicy { get { throw null; } set { } }
```
[the remark this answers is not in the document]
Comment (angiurgiu) [reply sdk/communication/Azure.Communication.Chat/api/Azure.Communication.Chat.netstandard2.0.cs:287]: Yes, the clients need to be able to set UpdateChatThreadPropertiesOptions.RetentionPolicy before passing it onto the Update API

The topic in this same class also has a setter

File: sdk/communication/Azure.Communication.Chat/src/autorest.md (hunk 2 above)
Comment (angiurgiu) [reply sdk/communication/Azure.Communication.Chat/src/autorest.md:17]: Updated

**Review by LuChen-Microsoft — APPROVED** [review]

File: sdk/communication/Azure.Communication.CallAutomation/src/autorest.md [hunk 5]
```diff
@@ -10,7 +10,7 @@ model-namespace: false
 tag: package-2023-10-03-preview
 
 require:
-    - https://github.com/Azure/azure-rest-api-specs/blob/2f0c2b15ba41419aec1a739307b50e03178e0d4d/specification/communication/data-plane/CallAutomation/readme.md
+    - https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/communication/data-plane/Chat/preview/2024-03-15-preview/communicationserviceschat.json
```
Comment (LuChen-Microsoft) [inline sdk/communication/Azure.Communication.CallAutomation/src/autorest.md:13]: .NET build asks for a specific commit version: https://[raw.githubusercontent.com/Azure/azure-rest-api-specs/3fb73ef5a3af2c138b53e3cced182095b671a679/specification/communication/data-plane/Chat/readme.md](https://raw.githubusercontent.com/Azure/azure-rest-api-specs/3fb73ef5a3af2c138b53e3cced182095b671a679/specification/communication/data-plane/Chat/readme.md)

**Pull request merged**